pdcupdater/handlers/depchain/rpms.py

```diff
@@ -0,0 +1,249 @@
+        time.sleep(1)
+
+        # First, go through all of the relationships that we learn from koji,
+        # and add them to PDC.  Some may already be present, but we may add new
+        # ones here.
+        koji_relationships = self.get_koji_relationships_from_build(
+            self.koji_url, msg['msg']['build_id'])
+        for type, child in koji_relationships:
+            child = pdcupdater.utils.ensure_release_component_exists(
+                pdc, release_id, child)
+            pdcupdater.utils.ensure_release_component_relationship_exists(
+                pdc, parent=parent, child=child, type=type)
+
+        # Lastly, go through all of the relationships that we know of now in
+        # PDC and find any that do not appear in koji.  These must be old
+        # relationships that are no longer relevant.
+        pdc_relationships = list(self._yield_pdc_relationships(pdc, release_id))
+        for entry in pdc_relationships:
+            pdcupdater.utils.delete_release_component_relationship(
+                pdc, parent=parent, child=child, type=type)
+
+    def audit(self, pdc):
+        present, absent = set(), set()
+        tags = interesting_tags()
+
+        for tag in tags:
+            log.debug("Starting audit of tag %r." % tag)
+            release_id, release = tag2release(tag)
+
+            # Query Koji and PDC to figure out their respective opinions
+            koji_relationships = list(self._yield_koji_relationships(pdc, tag))
+            pdc_relationships = list(self._yield_pdc_relationships(pdc, release_id))
+
+            # Filter out any irrelevant relationships (those of some type
+            # managed by a different pdc updater Handler).
+            pdc_relationships = [
+                entry for entry in pdc_relationships
+                if entry['type'] in self.managed_types
+            ]
+
+            # normalize the two lists
+            koji_relationships = set(["%s/%s %s %s/%s" % (
+                parent['name'],
+                parent['release']['release_id'],
+                relationship,
+                child['name'],
+                child['release']['release_id'],
+            ) for parent, relationship, child in koji_relationships])
+            pdc_relationships = set(["%s/%s %s %s/%s" % (
+                entry['from_component']['name'],
+                entry['from_component']['release'],
+                entry['type'],
+                entry['to_component']['name'],
+                entry['to_component']['release'],
+            ) for entry in pdc_relationships])
+
+            # use set operators to determine the difference
+            present = present.union(pdc_relationships - koji_relationships)
+            absent = absent.union(koji_relationships - pdc_relationships)
+
+        return present, absent
+
+    def initialize(self, pdc):
+        tags = interesting_tags()
+
+        for tag in tags:
+            log.debug("Starting audit of tag %r." % tag)
```

Comment:
I don't think this was meant to be here. If it is, a docstring clarifying what this function is trying to do would be very helpful.